staging/src/k8s.io/client-go/transport/transport.go

```diff
@@ -56,6 +60,22 @@ func New(config *Config) (http.RoundTripper, error) {
 	return HTTPWrappersForConfig(config, rt)
 }
 
+func isValidHolders(config *Config) bool {
```

Comment:
expand this to make sure the address of the functions held in the holders match the functions

Reply:
`c.HasCertCallback() ` should be updated to use this helper too, right now only checks for 
```
// HasCertCallback returns whether the configuration has certificate callback or not.
func (c *Config) HasCertCallback() bool {
	return c.TLS.GetCert != nil
}
```

Reply:
just checking for GetCert is sufficient if that is the effective field, and GetCertHolder is just used for caching

Reply:
ah, I see, we use this validation to assert that those fields are always set ... 

> expand this to make sure the address of the functions held in the holders match the functions

I understand know this comment, this is what makes this condition necessary and enough to guarantee everything works  :+1: